src/Pages/SignUp/SignUp.jsx

```diff
@@ -0,0 +1,347 @@
+      })
+    } else {
+      this.setState({
+        birthYearValidationMassage: '',
+      })
+    }
+  }
+
+  handleBirthMonthValidation = () => {
+    const checkBirthMonth = this.state.birthMonth.length === 2
+    const checkBirthMonth2 = this.state.birthMonth <= 12
+    if (!checkBirthMonth || !checkBirthMonth2) {
+      this.setState({
+        birthMonthValidationMassage: '태어난 월 2자리를 정확하게 입력하세요.',
+      })
+    } else {
+      this.setState({
+        birthMonthValidationMassage: '',
+      })
+    }
+  }
+
+  handleBirthDayValidation = () => {
+    const checkBirthDay = this.state.birthDay.length === 2
+    const checkBirthDay2 = this.state.birthDay <= 31
+    if (!checkBirthDay || !checkBirthDay2) {
+      this.setState({
+        birthDayValidationMassage:
+          '태어난 일(날짜) 2자리를 정확하게 입력하세요.',
+      })
+    } else {
+      this.setState({
+        birthDayValidationMassage: '',
+      })
+    }
+  }
+
+  handleGenderChange = (event) => {
+    this.setState({
+      gender: event.target.value,
+    })
+    if (event.target.value === 'none') {
+      this.setState({
+        genderValidationMassage: '필수 정보입니다.',
+      })
+    } else {
+      this.setState({
+        genderValidationMassage: '',
+      })
+    }
+  }
+
+  render() {
+    return (
+      <div className='signUp'>
+        <header>
+          <img
+            className='logoImg'
+            alt='logo'
+            src='/images/line-amigos-logo.png'
+          />
+        </header>
+        <div className='IdPwForm'>
+          <div className='IdBox'>
+            <span className='label'>아이디</span>
+            <input
+              className='IdPwBox'
+              id='email'
+              type='text'
+              placeholder='@lineamigos.com'
+              value={this.state.email}
+              onChange={this.handleValueChange}
+              onKeyUp={this.handleEmailValidation}
+            />
+            <span className='validationMassage'>
+              {this.state.emailValidationMassage}
+            </span>
+          </div>
+          <div className='pwBox'>
+            <span className='label'>비밀번호</span>
+            <input
+              className='IdPwBox'
+              id='pw'
+              type='password'
+              value={this.state.pw}
+              onChange={this.handleValueChange}
+              onKeyUp={this.handlePwValidation}
+            />
+            <span className='validationMassage'>
+              {this.state.pwValidationMassage}
+            </span>
+          </div>
+          <div className='rePwBox'>
+            <span className='label'>비밀번호 재확인</span>
+            <input
+              className='IdPwBox'
+              id='rePw'
+              type='password'
+              value={this.state.rePw}
+              onChange={this.handleValueChange}
+              onKeyUp={this.handleRePwValidation}
+            />
+            <span className='validationMassage'>
+              {this.state.rePwValidationMassage}
```

Comment:
handleAllValidation 함수와 각각의 유효성 함수가 실행될 필요 없이 
예를 들어 state에서 isValid 라는 값을 미리 지정하고 가입하기 버튼을 눌렀을때 isValid가 true로 바뀌면서 isValid인 조건을 넣어서 각각의 메세지를 띄워주는 방식으로 고쳐주시는게 나을것 같습니다. 아래의 코드를 참고해주세요. 
미현님 다른 함수들도 마찬가지로 수정해보세요~ 

```suggestion
           {this.state.isValid && (this.state.rePw !== this.state.pw) ? '비밀번호가 일치하지 않습니다.' : ' ' }
```

Reply:
말씀해주신대로 isValid 사용하여 함수 수정완료했습니다. 

` {
                (this.state.isValid && !this.state.pw ? '필수 정보입니다.' : '',
                this.state.isValid && !pwValidation.test(this.state.pw)
                  ? '8~16자 영문 대/소문자, 숫자, 특수문자를 모두 사용하세요.'
                  : '')
              }` 

다만, 이렇게 고치고 나니 가입하기 버튼 누르기 전 바로 input창에 입력했을 때는 경고창이안나옵니다.. 혹은 아래와 같이 handleChange가 발생하고 바로 isValid가 발생하도록 하니 input창 하나만 입력해도 모든 창에 경고가 뜹니다.. 흑, 제 머릿속으로는 이러면 다시 여러 함수를 적어주는 방법들만 떠오르는데... 

`handleValueChange = (event) => {
    const { id, value } = event.target
    this.setState({
      [id]: value,
    })
    this.handleIsValid()
  }`